lib/jsonapi/serializer.ex

```diff
@@ -43,11 +43,9 @@ defmodule JSONAPI.Serializer do
       id: view.id(data),
       type: view.type(),
       attributes: underscore(view.attributes(data, conn)),
-      relationships: %{},
-      links: %{
-        self: view.url_for(data, conn)
-      }
+      relationships: %{}
     }
+    |> merge_links(data, view, conn, Application.get_env(:jsonapi, :remove_links, false))
```

Comment:
This is a little funky.  Do you think we could capture the above map and add it as the first argument rather than using a pipe?  For one, we don't need pipes when there's a single pipe and it's a bit different to see a pipe setup like this.